src/include/btree_inline.h

```diff
@@ -1793,41 +1793,20 @@ static inline bool
 __wt_page_evict_retry(WT_SESSION_IMPL *session, WT_PAGE *page)
 {
     WT_PAGE_MODIFY *mod;
-    WT_TXN_GLOBAL *txn_global;
-    wt_timestamp_t pinned_ts;
 
-    txn_global = &S2C(session)->txn_global;
-
-    /*
-     * If the page hasn't been through one round of update/restore, give it a try.
-     */
-    if ((mod = page->modify) == NULL || !FLD_ISSET(mod->restore_state, WT_PAGE_RS_RESTORED))
```

Comment:
I removed `!FLD_ISSET(mod->restore_state, WT_PAGE_RS_RESTORED)` because I don't think it is that useful anymore. If we have done update restore, whether we need to retry or not depends on this check `(mod->last_evict_pass_gen + 5 < S2C(session)->cache->evict_pass_gen)`. If we haven't done update restore, `(mod->last_evict_pass_gen + 5 < S2C(session)->cache->evict_pass_gen)` will likely be true anyway.

Reply:
If you still think this check is useful, I can put it back.

Reply:
I put this check back.